src/backend/joanie/core/models/products.py

```diff
@@ -85,6 +87,27 @@ def __str__(self):
             f"{self.price}"
         )
 
+    @property
+    def target_course_runs(self):
+        """
+        Retrieve all course runs related to the product instance.
+
+        We select all course runs linked to the product course relation or related to
+        the product target courses which course relation does not have course runs.
```

Comment:
This sentence did not help me understand and it took me some time to convince myself that your filters are correct.
I propose:

Returns: 
    - all course runs related to target courses for which the product/course relation does not specify a list of eligible course runs (see course_runs field on the ProductCourseRelation model)
    - only the course runs specified on the product/course relation for target courses  on which no list of eligible course runs was specified on the product/course relation. 
    -

Reply:
All right! It's clear.